example/lib/main.dart

```diff
@@ -60,15 +60,14 @@ class _HomeState extends State<Home> {
       "totalSamples": totalSamples,
     };
     final samplesData = await compute(loadparseJson, audioDataMap);
-    await audioPlayer.load(audioData[1]);
-    await audioPlayer.play(audioData[1]);
+    await audioPlayer.audioCache.load(audioData[1]);
+    // await audioPlayer.audioCache(audioData[1]);
     // maxDuration in milliseconds
     await Future.delayed(const Duration(milliseconds: 200));
 
-    int maxDurationInmilliseconds =
-        await audioPlayer.fixedPlayer!.getDuration();
+        await audioPlayer.getDuration().then((value) =>
+        maxDuration = value!);
```

Comment:
It's not safe to bang! A better approach would be to set the default value to 0.

```dart
maxDuration = value ?? Duration.zero
```

Reply:
fixed